packages/core/src/com/communication.ts

```diff
@@ -1,182 +1,182 @@
+    DuplicateRegistrationError,
+    EnvironmentDisconnectedError,
+    UnConfiguredMethodError,
+    UnknownCallbackIdError
+} from './communication-errors';
 
 /**
  * Manages all API registrations and message forwarding
  * in each execution context.
  */
 export class Communication {
-    private rootEnvId: string;
-    private rootEnvName: string;
-    private idsCounter = new MultiCounter();
-    private disposing = false;
+    static DEBUG = false;
+    private readonly idsCounter = new MultiCounter();
     private readonly callbackTimeout = 60_000 * 5; // 5 minutes
     private readonly slowThreshold = 5_000; // 5 seconds
-    private pendingEnvs: SetMultiMap<string, () => void> = new SetMultiMap();
-    private pendingMessages = new SetMultiMap<string, UnknownFunction>();
-    private handlers = new Map<string, Set<UnknownFunction>>();
-    private eventDispatchers: { [dispatcherId: string]: SerializableMethod } = {};
-    private apis: RemoteAPIServicesMapping = {};
-    private apisOverrides: RemoteAPIServicesMapping = {};
-    private options: Required<CommunicationOptions>;
-    private readyEnvs = new Set<string>();
-    private environments: { [environmentId: string]: EnvironmentRecord } = {};
-    private messageHandlers = new WeakMap<Target, (options: { data: null | Message }) => void>();
-    private disposeListeners = new Set<(envId: string) => void>();
-    private pendingCallbacks = new Map<string, CallbackRecord<unknown>>();
-    private messageIdPrefix: string;
-    // manual DEBUG_MODE
-    private DEBUG = false;
+    private readonly pendingEnvs = new SetMultiMap<string, () => void>();
+    private readonly pendingMessages = new SetMultiMap<string, UnknownFunction>();
+    private readonly handlers = new Map<string, Set<UnknownFunction>>();
+    private readonly eventDispatchers: { [dispatcherId: string]: SerializableMethod } = {};
+    private readonly apis: { [remoteServiceId: string]: Record<string, AnyFunction> } = {};
+    private readonly apisOverrides: typeof this.apis = {};
+    private readonly readyEnvs = new Set<string>();
+    private readonly environments: { [environmentId: string]: EnvironmentRecord } = {};
+    private readonly messageHandlers = new WeakMap<Target, (options: { data: null | Message }) => void>();
+    private readonly disposeListeners = new Set<(envId: string) => void>();
+    private readonly pendingCallbacks = new Map<string, CallbackRecord<unknown>>();
+    private readonly callbackIdPrefix: string;
+    private readonly options: {
+        warnOnSlow: boolean;
+        publicPath: string;
+        connectedEnvironments: { [environmentId: string]: ConfigEnvironmentRecord };
+    } = {
+        warnOnSlow: Communication.DEBUG,
+        publicPath: '',
+        connectedEnvironments: {},
+    };
+    private disposing = false;
+
     constructor(
-        private host: Target,
-        id: string,
+        private readonly host: Target,
+        private readonly id: string,
```

Comment:
this is so confusing that we still have `rootEnvId` and this `id` now. choose one.

Reply:
I cannot follow if this change is valid because of that

Reply:
OK, `rootEnvId` gone in favor of just `id`